frame/support/src/storage/mod.rs

```diff
@@ -563,6 +572,138 @@ impl<T> Iterator for PrefixIterator<T> {
 	}
 }
 
+/// Iterate over a prefix of a child trie and decode raw_key and raw_value into `T`.
+///
+/// If any decoding fails it skips the key and continues to the next one.
+pub struct ChildTriePrefixIterator<T> {
+	/// The prefix iterated on
+	prefix: Vec<u8>,
+	/// Storage key of the child trie
+	storage_key: Vec<u8>,
+	/// The last key iterated on
+	previous_key: Vec<u8>,
+	/// If true then values are removed while iterating
+	drain: bool,
+	/// Whether or not the first key i.e. the prefix is fetched
```

Comment:
I see why you did that, but first key is not always the prefix.

If think it would make more sense to rename it fetch_previous_key.
And we can remove the function `fetch_prefix` and just always set `fetch_previous_key` to true when we create it.
If somebody needs to skip the first key, then can call next or we can add other function to do `iter_prefix_starting_at(prefix, starting_key)`